packages/rs-dpp/src/state_transition/state_transition_facade.rs

```diff
@@ -0,0 +1,425 @@
+use crate::identity::state_transition::identity_create_transition::validation::basic::IdentityCreateTransitionBasicValidator;
+use crate::identity::state_transition::identity_credit_withdrawal_transition::validation::basic::validate_identity_credit_withdrawal_transition_basic::IdentityCreditWithdrawalTransitionBasicValidator;
+use crate::identity::state_transition::identity_topup_transition::validation::basic::IdentityTopUpTransitionBasicValidator;
+use crate::identity::state_transition::identity_update_transition::validate_identity_update_transition_basic::ValidateIdentityUpdateTransitionBasic;
+use crate::identity::state_transition::validate_public_key_signatures::PublicKeysSignaturesValidator;
+use crate::identity::validation::PublicKeysValidator;
+
+use crate::state_repository::StateRepositoryLike;
+use crate::state_transition::{
+    StateTransition, StateTransitionFactory, StateTransitionFactoryOptions,
+};
+
+use crate::state_transition::state_transition_execution_context::StateTransitionExecutionContext;
+use crate::state_transition::validation::validate_state_transition_basic::StateTransitionBasicValidator;
+use crate::state_transition::validation::validate_state_transition_by_type::StateTransitionByTypeValidator;
+use crate::state_transition::validation::validate_state_transition_fee::StateTransitionFeeValidator;
+use crate::state_transition::validation::validate_state_transition_identity_signature::validate_state_transition_identity_signature;
+use crate::state_transition::validation::validate_state_transition_key_signature::StateTransitionKeySignatureValidator;
+use crate::state_transition::validation::validate_state_transition_state::StateTransitionStateValidator;
+use crate::validation::{
+    AsyncDataValidator, AsyncDataValidatorWithContext, SimpleValidationResult,
+};
+use crate::version::ProtocolVersionValidator;
+
+#[derive(Clone)]
+pub struct StateTransitionFacade<SR, BLS>
+where
+    SR: StateRepositoryLike + Clone,
+    BLS: BlsModule + Clone,
+{
+    state_repository: Arc<SR>,
+    basic_validator:
+        Arc<StateTransitionBasicValidator<SR, StateTransitionByTypeValidator<SR, BLS>>>,
+    state_validator: Arc<StateTransitionStateValidator<SR>>,
+    key_signature_validator: Arc<StateTransitionKeySignatureValidator<SR>>,
+    fee_validator: Arc<StateTransitionFeeValidator<SR>>,
+    bls: BLS,
+    factory: Arc<StateTransitionFactory<SR, BLS>>,
+}
+
+impl<SR, BLS> StateTransitionFacade<SR, BLS>
+where
+    SR: StateRepositoryLike + Clone,
+    BLS: BlsModule + Clone,
+{
+    pub fn new(
+        state_repository: SR,
+        adapter: BLS,
+        protocol_version_validator: Arc<ProtocolVersionValidator>,
+    ) -> Result<Self, ProtocolError> {
+        let wrapped_state_repository = Arc::new(state_repository.clone());
+
+        let state_transition_basic_validator;
```

Comment:
I'd suggest instead of doing this and then initializing it in the code blocks later if you want to use code blocks initialize the variable at the same place as the code block, i.e.: `let state_transition_basic_validator = {...;  StateTransitionFactory::new(...)};`